From 2be0856e09f47e7f2c35e77d4e000d769c6beaf9 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 11 Jul 2017 10:18:13 +0200
Subject: [PATCH 4/4] chromium: WebKit fix build with gcc7

Based on qtwebengine-opensource-src-5.8.0-wtf-gcc7.patch from
http://git.net/ml/scm-fedora-commits/2017-03/msg15697.html

* Fixes:
In file included from ../../git/src/third_party/WebKit/Source/platform/heap/ThreadingTraits.h:14:0,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/ThreadState.h:37,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/GarbageCollected.h:8,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/Visitor.h:35,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/GCInfo.h:8,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/Heap.h:35,
                 from ../../git/src/third_party/WebKit/Source/platform/heap/Handle.h:34,
                 from ../../git/src/third_party/WebKit/public/platform/WebPrivatePtr.h:38,
                 from ../../git/src/third_party/WebKit/public/platform/WebString.h:35,
                 from ../../git/src/third_party/WebKit/Source/platform/exported/FilePathConversion.cpp:8:
../../git/src/third_party/WebKit/Source/wtf/LinkedHashSet.h: In member function 'void WTF::LinkedHashSet<Value, HashFunctions, HashTraits, Allocator>::swap(WTF::LinkedHashSet<Value, HashFunctions, HashTraits, Allocator>&)':
../../git/src/third_party/WebKit/Source/wtf/LinkedHashSet.h:549:5: error: there are no arguments to 'swapAnchor' that depend on a template parameter, so a declaration of 'swapAnchor' must be available [-fpermissive]
     swapAnchor(m_anchor, other.m_anchor);
     ^~~~~~~~~~
../../git/src/third_party/WebKit/Source/wtf/LinkedHashSet.h:549:5: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
---
 chromium/third_party/WebKit/Source/wtf/LinkedHashSet.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/chromium/third_party/WebKit/Source/wtf/LinkedHashSet.h b/chromium/third_party/WebKit/Source/wtf/LinkedHashSet.h
index e85c72fd65..6f94cd6734 100644
--- a/chromium/third_party/WebKit/Source/wtf/LinkedHashSet.h
+++ b/chromium/third_party/WebKit/Source/wtf/LinkedHashSet.h
@@ -542,6 +542,8 @@ inline LinkedHashSet<T, U, V, W>& LinkedHashSet<T, U, V, W>::operator=(LinkedHas
     return *this;
 }
 
+inline void swapAnchor(LinkedHashSetNodeBase& a, LinkedHashSetNodeBase& b);
+
 template<typename T, typename U, typename V, typename W>
 inline void LinkedHashSet<T, U, V, W>::swap(LinkedHashSet& other)
 {
-- 
2.13.2

